### DISCUSSION TITLE
How to base styles on WinUI 3 styles

### DISCUSSION THREAD

**Discussion by DavidM29** [unoplatform/uno#17774] [Q&A] [answered]
Hello, I noticed that if I create styles and do not base them on any other style, the controls do not take the WinUI 3 styles. As I am relatively new to WinUI and uno I am not sure if I am doing something wrong or if that is the intended behavior.

As I read some articles about this, I think I have to base my styles on the default styles, however the default styles are not found when I am trying to reference them as static resources. Is this maybe related to #4424 or am I just doing something wrong?

**Comment by kazo0** [comment]
Hi @DavidM29!

Would you be able to upload a sample app .zip where you are experiencing this issue? I can check to see if anything needs to be changed

**Reply by DavidM29** [reply]
So, I am not sure what I did wrong last time I checked this, however, as I was making the sample I made it work. There is a chance I did not test it well because when I reference "DefaultButtonStyle" as a static resource a waning pops up that it can not be found, but when I run the application it works.
It seems I definitely was doing something wrong, thanks for the help though.

**Reply by kazo0** [reply]
Hmm interesting, let me know if you happen to run into it again. Seems curious!

**Reply by DavidM29** [reply]
Hey @kazo0 , I encountered this issue again.

Apparently, some default styles can not be found when running on windows, but are available on android (I am not sure about other platforms, I have not tested them).

Just ran into an exception when running a packaged app on windows: 
```
The text associated with this error code could not be found.

Cannot find a Resource with the Name/Key DefaultExpanderStyle [Line: 22 Position: 13]
```
However, when running the same app on android I do not get this exception.

I encountered this because, on android, when I declare a style for an expander, if I do not base it on the default style, the expander is broken, however I do not need to base the style on the default one on windows.

I am not sure if this is expected behavior or a bug.

**Reply by kazo0** [reply]
@DavidM29 if you are still experiencing this, would you be able to upload the .zip of that project?

**Comment by DavidM29** [comment]
[UnoApp1.zip](https://github.com/user-attachments/files/16862136/UnoApp1.zip)
Here is this minimal project reproducing this issue.
If you run it on android, there are no issues, it works as expected, however, if you run it on windows it throws the exception I mentioned above.

If the `BasedOn` property of the style is not specified, everything works as expected on windows, however, only the content of the expander gets rendered on android.

**Reply by kazo0** [reply, marked as answer]
Seems like an issue with winui: https://github.com/microsoft/microsoft-ui-xaml/issues/6528

The suggestion in that issue is to remove the `BasedOn`. Curious that it is required for non-Windows platforms. 

For now, you can use [platform conditionals](https://platform.uno/docs/articles/platform-specific-xaml.html) and define this in your xaml

```xml
xmlns:not_win="http://uno.ui/not_win"
mc:Ignorable="not_win"
```

and use this for BasedOn:

```xml
not_win:BasedOn="{StaticResource DefaultExpanderStyle}"
```

I will create an issue in the Uno repo to investigate the need for the BasedOn in Uno platforms

**Reply by kazo0** [reply]
https://github.com/unoplatform/uno/issues/18115

**Reply by DavidM29** [reply]
Thanks for the answer,

> For now, you can use [platform conditionals](https://platform.uno/docs/articles/platform-specific-xaml.html) and define this in your xaml

That is what I am doing at the moment.


> I will create an issue in the Uno repo to investigate the need for the BasedOn in Uno platforms

Do I close this discussion?

**Reply by kazo0** [reply]
@DavidM29 You shouldn't need to do anything for this discussion, as long as there is a post that is marked as the answer then it should be fine :)

I can re-ping you once we find out the issue for the styles and we have a version available with the fix
